### PR TITLE
feat: added method for setting blocklist with relevant tests

### PR DESCRIPTION
**Pull request by Nilay27**
* Adding method to set blocklist for accounts using the `set(address[] calldata _accounts, bool[] calldata _blocked)` 
* Added relevant tests for checking the following blocking/unblocking scenarios:
  1. Blocking a single account
  2. Blocking multiple accounts
  3. Unblocking a single account
  4. Unblocking multiple accounts
  5. Blocking and unblocking accounts in a single call (i.e blocking one account while unblocking another)

### CODE REVIEW

File: apps/smart-contracts/token/test/Blocklist.test.ts [hunk 1]
```diff
@@ -38,4 +49,103 @@ describe('=> Blocklist', () => {
...
+      for (let i = 0; i < blockedUsersArray.length; i++) {
+        expect(await blocklist.isAccountBlocked(blockedUsersArray[i])).to.eq(false)
+      }
+
+      await blocklist.connect(owner).set(blockedUsersArray, blockedArray)
+
+      for (let i = 0; i < blockedUsersArray.length; i++) {
+        expect(await blocklist.isAccountBlocked(blockedUsersArray[i])).to.eq(true)
+      }
+
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/test/Blocklist.test.ts:139]:
```suggestion
```

Not sure why you are checking this twice? Seems to be a mistake.

**Review by ramenforbreakfast — APPROVED** [review]
LGTM, besides one comment

File: apps/smart-contracts/token/test/Blocklist.test.ts [hunk 2]
```diff
@@ -38,4 +49,103 @@ describe('=> Blocklist', () => {
       expect(await blocklist.getBlockedAllountsIndex()).to.eq(0)
     })
   })
+
+  describe('# set', () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/Blocklist.test.ts:53]: add test for when the same account is passed in multiple times within the array
(should set as the lattermost bool value)

File: apps/smart-contracts/token/test/Blocklist.test.ts [hunk 3]
```diff
@@ -38,4 +49,103 @@ describe('=> Blocklist', () => {
...
+    })
+
+    it('blocks and unblocks accounts', async () => {
+      await blocklist.connect(owner).set([unblockedUser1.address], [true])
+      expect(await blocklist.isAccountBlocked(unblockedUser1.address)).to.eq(true)
+      expect(await blocklist.isAccountBlocked(blockedUser1.address)).to.eq(false)
+
+      await blocklist
+        .connect(owner)
+        .set([blockedUser1.address, unblockedUser1.address], [true, false])
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/Blocklist.test.ts:123]: nit: switch the order here so it matches the order in the expects

File: apps/smart-contracts/token/test/Blocklist.test.ts [hunk 4]
```diff
@@ -1,22 +1,33 @@
+/* eslint-disable no-await-in-loop */
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/Blocklist.test.ts:1]: why?

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

**Pull request closed without merging**